airflow/triggers/temporal.py

```diff
@@ -41,10 +41,8 @@ def __init__(self, moment: datetime.datetime):
         # Make sure it's in UTC
         elif moment.tzinfo is None:
             raise ValueError("You cannot pass naive datetimes")
-        elif not hasattr(moment.tzinfo, "offset") or moment.tzinfo.offset != 0:  # type: ignore
-            raise ValueError(f"The passed datetime must be using Pendulum's UTC, not {moment.tzinfo!r}")
         else:
-            self.moment = moment
+            self.moment = timezone.convert_to_utc(moment)
```

Comment:
"Tests Police" here 🤣 , we need also test this one.

A guess we do not have unittests for this constructor except type checking.
The same valid for `TimeDeltaTrigger` which based on `DateTimeTrigger`

Reply:
I just updated the test `test_datetime_trigger_timing` to check if the logic remains valid with different timezones

Reply:
I guess we could check just in class constructor but this seems like even better

Reply:
@Taragolis looks like there are tests added. Do you think there should be more testing?